src/test/java/programminglife/util/NumberOnlyListenerTest.java

```diff
@@ -0,0 +1,47 @@
+package programminglife.util;
+
+import javafx.scene.control.TextField;
+import javafx.scene.layout.AnchorPane;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import programminglife.utility.InitFXThread;
+import programminglife.utility.NumbersOnlyListener;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Created by iwanh on 14/06/2017.
+ */
+public class NumberOnlyListenerTest {
+
+    TextField tf;
+
+    @BeforeClass
+    public static void setUpClass() throws Exception {
+        InitFXThread.setupClass();
+    }
+
+    @Before
+    public void setUp(){
+        tf = new TextField();
+        tf.textProperty().addListener(new NumbersOnlyListener(tf));
```

Comment:
Well, the `Listener` requires the `TextField`, so there is no other way (except mocking).